packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -355,6 +356,18 @@ class _ZoomEnterTransitionState extends State<_ZoomEnterTransition> with _ZoomTr
     super.didUpdateWidget(oldWidget);
   }
 
+  @override
+  void didChangeDependencies() {
+    // If the screen size changes during the transition, perhaps due to
+    // a keyboard dismissal, then ensure that contents are re-rasterized once.
+    final MediaQueryData? data = MediaQuery.maybeOf(context);
+    if (mediaQueryData?.size != data?.size) {
```

Comment:
nit: if we only care about size, just cache the size?

Reply:
Also, I should probably know this because I worked on it at one point, but isn't the IME "just" changing the insets/padding values and not the size?

I tried to come up with an IME based reproduction for the internal bug that could be used in open source and failed, but I hadn't spent very long on it. If you've confirmed this fixes the internal bug that's cool, but if it hasn't been confirmed internally it may be worth looking into and figuring out if we're being affected by insets/padding rather than size.

Reply:
See https://api.flutter.dev/flutter/widgets/MediaQueryData-class.html - I'm pretty sure it's not the size that'll get impacted here, but the viewInsets.

Reply:
Good idea, converting to draft for now. I think we should probably handle any of size and/or view insets. But particularly we care about the render object size, so maybe I should just check there instead

Reply:
Alternatively, we could just clear whenever anything on the media query changes, which runs a small risk of changing for things not visually impacting the page size.